server/src/main/java/org/apache/cassandra/sidecar/datahub/SchemaReporter.java

```
@@ -96,44 +110,72 @@ public SchemaReporter(@NotNull IdentifiersProvider identifiersProvider,
         this.tableConverters = tableConverters;
         this.emitterFactory = emitterFactory;
+        this.reportingMetrics = reportingMetrics;
     }
 
     /**
-     * Public method for converting and reporting the Cassandra schema
+     * Public method for converting and reporting the Cassandra schema when triggered by a scheduled periodic task
      *
      * @param cluster the {@link Cluster} to extract Cassandra schema from
      */
-    public void process(@NotNull Cluster cluster)
+    public void processScheduled(@NotNull Cluster cluster)
+    {
+        process(cluster.getMetadata(), reportingMetrics.startedSchedule.metric);
+    }
+
+    /**
+     * Public method for converting and reporting the Cassandra schema when triggered by a received API request
+     *
+     * @param metadata the {@link Metadata} to extract Cassandra schema from
+     */
+    public void processRequested(@NotNull Metadata metadata)
     {
-        process(cluster.getMetadata());
+        process(metadata, reportingMetrics.startedRequest.metric);
     }
 
     /**
-     * Public method for converting and reporting the Cassandra schema
+     * Private method for converting and reporting the Cassandra schema
      *
      * @param metadata the {@link Metadata} to extract Cassandra schema from
+     * @param started the {@link DeltaGauge} for the metric counting invocations
      */
-    public void process(@NotNull Metadata metadata)
+    private void process(@NotNull Metadata metadata,
+                         @NotNull DeltaGauge started)
     {
+        LOGGER.info("Starting to report schema for cluster, identifiers={}", identifiersProvider);
+        started.increment();
+
         try (Emitter emitter = emitterFactory.emitter())
         {
-            stream(metadata).forEach(ThrowableUtils.consumer(emitter::emit));
+            Stopwatch stopwatch = Stopwatch.createStarted();
+            long counter = stream(metadata).map(ThrowableUtils.function(emitter::emit))
```

This count seems to ignore exceptions in emit, what will we be using this counter metric for?

---

Exception in emit will be wrapped into a `RuntimeException` and rethrown, aborting conversion, and resulting in the `Failed` metric being updated from the `catch`. I guess we don't care about the number of aspects produced before failure has happened…